### ISSUE TITLE
Beam: add possibility to specify conf for remote adapters

### ISSUE THREAD

**Issue by chuwy** [snowplow/enrich#271]
Remote adapters require some configuration, for example : 
```
remoteAdapters = [
￼	   {
￼	       vendor: "special-vendor"
￼	       version: "v1"
￼	       url: "http://remote-adapter.com:9090"
￼	       connectionTimeout: 1000
￼	       readTimeout: 5000
￼	   }
￼]
```
Currently there is no way to pass this configuration to `beam-enrich`, as it only accepts command-line arguments.

A solution could be to add an optional argument containing the path to the configuration file with the remote adapters.

Moved from: https://github.com/snowplow/snowplow/issues/4080

**Comment by chuwy** [comment]
Migrated from https://github.com/snowplow/beam-enrich/issues/24 (comments are auto-generated)

**Comment by chuwy** [comment]
Copying over from #310:

As side job we can:

1. Move [predefined adapters](https://github.com/snowplow/enrich/blob/bf94b7365a09003e9baecdb9844a787324e3cfad/modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/AdapterRegistry.scala#L39-L59) into a companion object 
2. Use something like `CollectorPayload.Api` instead of `(String, String)`
3. Remove excessive use of default parameters everywhere
4. Define configuration case classes for remote adapters in `common`

**Comment by JonMerlevede** [comment]
While this is worked on, is there any way to use remote adapters with Beam Enrich?

**Comment by chuwy** [comment]
I don't think there are, @JonMerlevede unless you want to assemble Beam Enrich yourself - I can guide you if you post on our [forums](discourse.snowplowanalytics.com/) (we use Github for bugreports and feature requests only).

However, out of curiosity what is your use case? I'm asking because to me remote adapters seem like a bit controversial feature and I've been even considering removing it in a very distant future.

**Comment by JonMerlevede** [comment]
Thanks for the response! It's good to know that remote adapters are controversial; from the docs I gathered they were the recommended way for ingesting unsupported event types. I see several other ways to achieve my goal, including writing and contributing an adapter in Scala. As you suggested, I created a topic on this here: https://discourse.snowplowanalytics.com/t/ingesting-auth0-log-events/4470.
